packages/spatial/src/renderer/components/FogSettingsComponent.ts

```diff
@@ -90,6 +93,17 @@ export const FogSettingsComponent = defineComponent({
   reactor: () => {
     const entity = useEntityContext()
     const fog = useComponent(entity, FogSettingsComponent)
+    const isVisible = useOptionalComponent(entity, VisibleComponent)
+    const retriggerFogType = useHookstate(0)
+
+    useEffect(() => {
```

Comment:
i think rather than the multiple condition based logic we already have and are extending here, we should make the main useEffect with the switch statement have it's effects be reversable, using `return () => {}` to undo whatever the logic is in the switch statements

Reply:
ah I see the way to do this better now, thanks. I finally understood how to fix this after lots of testing and didn't realize the solution could be simplified